docs/v3tov4.md

```diff
@@ -0,0 +1,340 @@
+		}
+	}
+  ...
+```
+
+To add metadata just call 
+
+`func (i *Entity) AddMetadata(key string, value interface{})`
+
+To add a tag just call 
+
+`func (i *Entity) AddTag(key string, value interface{})`
+
+On both cases, if the `key` already exists the previous value will be overwritten with the new one.
+
+### Metrics
+
+#### Types
+
+In this SDK v4 we introduce some new metric types while removing some others.
+
+## Metric types in v4
+
+* Gauge
+
+  It is a value that can increase or decrease. It generally represents the value for something at a particular moment 
+  in time.
+  
+  Examples include the CPU load or the memory consumption.
+
+* Count
+
+  Measures the number of occurrences of an event given a time interval.
+
+  Examples include cache hits per reporting interval and the number of threads created per reporting interval.
+
+* Summary
+
+  Used to report pre-aggregated data, or information on aggregated discrete events. A summary 
+  includes a `count`, `sum` value, `min` value, and `max` value. 
+
+  Examples include transaction count/durations and queue count/ durations.
+  
+* Cumulative-count
+  
+  It generates a count metric by calculating the delta between the reported value and the previous one. Interval is 
+  calculated based on the timestamp of the datapoints. The value passed in is not a delta but instead the absolute value. 
+  
+  Examples include the total number of threads created for a process or the total number of cache hits since process 
+  startup.
+
+* Rate
+  
+  It generates a gauge metric by dividing the current value by the interval in seconds. Interval is calculated based on the 
+  timestamp of the datapoints.
+  
+  Examples include the number of get request per second.
+  
+* Cumulative-rate  
+
+  Similar to the rate, but this time we calculate the delta between the reported value and the previous one and then we
+  divide the result by the interval in seconds. Interval is calculated based on the timestamp of the datapoints.
+  
+  Examples include the total write bytes per second.
+
+#### Creating v4 metrics
+
+| metric            | method                                                                 |
+|-------------------|------------------------------------------------------------------------|
+| gauge             | integration.Gauge(timestamp, name, value)                              |
+| count             | integration.Count(timestamp, name, count)                              |
+| summary           | integration.Summary(timestamp, name, count, average, sum, min, max)    |
+| cumulative-count  | integration.CumulativeCount(timestamp, name, count)                    |
+| rate              | integration.Rate(timestamp, name, value)                               |
+| cumulative-rate   | integration.CumulativeRate(timestamp, name, value)                     |
+
+Description of the paramaters:
+
+`timestamp` (time.Time) is the metric's start time in Unix time (milliseconds).
```

Comment:
"start time" maybe not the most correct term. Maybe just say it's the timestamp of the metric/datapoint

Reply:
is the same explanation that's in here
https://docs.newrelic.com/docs/data-ingest-apis/get-data-new-relic/metric-api/report-metrics-metric-api#json-payload-keys